processors/baggage/baggagetrace/doc.go

```diff
@@ -2,4 +2,56 @@
 // SPDX-License-Identifier: Apache-2.0
 
 // Package baggagetrace implements a baggage span processor.
+//
+// This is an OpenTelemetry [Span Processor] that reads key/values stored in
+// [Baggage] in the starting span's parent context and adds them as attributes
+// to the span.
+//
+// Keys and values added to Baggage will appear on all subsequent child spans for
+// a trace within this service *and* will be propagated to external services via
+// propagation headers.
+// If the external services also have a Baggage span processor, the keys and
+// values will appear in those child spans as well.
+//
+// ⚠️ Waning ⚠️
+// To repeat: a consequence of adding data to Baggage is that the keys and values
+// will appear in all outgoing HTTP baggage headers from the application.
```

Comment:
typo + this is redundant

Reply:
I don't think this is redundant - it's a valid security concern that should be called out to ensure users understand the implication of using baggage. I can remove it if you feel strongly that it shouldn't be there though.

Reply:
> *and* will be propagated to external services via propagation headers.

This is already mentioned here. Maybe it could be reworded to make it more concerning?

>  keys and values will appear in all outgoing HTTP baggage headers from the application.

This is not correct. It would be only correct if e.g. `otelhttp` is used everywhere to make HTTP calls. Also baggage can appear in other outgoing traffic (e.g. message queues) in other form than HTTP headers.

Reply:
I think you're right - the warning may be a little heavy handed and does repeat itself a little. 

I've updated to remove the warning and repeated statement about HTTP propagation headers but left in the "don't put sensitive data in baggage".

What do you think of this? [b759d50](https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5619/commits/b759d50d85940684aacd88eebefda7e643083a8e)